Local storage prototype

jingxu97 opened the pull request:
User can specify a size limit for emptyDir volume. Eviction manager will
evict the pod if the usage of EmptyDIr volume exceeds the limit.

vishh committed:
adding local storage api types

Signed-off-by: Vishnu kannan <vishnuk@google.com>

vishh committed:
codegen

Signed-off-by: Vishnu kannan <vishnuk@google.com>

vishh committed:
protobuf generation

Signed-off-by: Vishnu kannan <vishnuk@google.com>

vishh committed:
generated code

Signed-off-by: Vishnu kannan <vishnuk@google.com>

msau42 committed:
Local volume plugin

msau42 committed:
LocalStorage API changes:
1. Add LocalStorage as a VolumeSource. It is required to implement the volume plugin.
2. Make PV.VolumeType optional
2. Add validation for LocalStorage

msau42 committed:
generated code

msau42 committed:
Update local storage API:
1. Remove unnecessary json from internal types.go
2. Add omitempty back to pvc.volumeType to make it optional.  But for
some reason the volumeType field is still required.  This needs more
investigation later.

Fix internal printers.
Add single deployment example.

msau42 committed:
Generated code

msau42 committed:
Update local PV example to use a real node name.

msau42 committed:
New scheduler predicate to filter prebound PVCs that use LocalStorage
PVs

vishh committed:
avoid handling local PVs in persistent volume controller

Signed-off-by: Vishnu kannan <vishnuk@google.com>

vishh committed:
avoid handling local PVs in persistent volume controller

Signed-off-by: Vishnu kannan <vishnuk@google.com>

msau42 committed:
GKE Local storage discovery daemonset
Add storage admin service account that runs the daemonset

msau42 committed:
[DO NOT MERGE] Auto-generated client-go files for new LocalStorage type

msau42 committed:
Update examples to use storage class
Add statefulset example

jingxu97 committed:
fix two compliing errors

jingxu97 committed:
Add Eviction when EmptyDir exceeds user's requested limit

User can specify a size limit for emptyDir volume. Eviction manager will
evict the pod if the usage of EmptyDIr volume exceeds the limit.

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/43333)
<!-- Reviewable:end -->

vishh committed:
Scheduler can now bind local PVs

Signed-off-by: Vishnu Kannan <vishnuk@google.com>

msau42 commented:
This is just a prototype and definitely not ready for any sort of review.  Can someone add the do not merge label?

msau42 committed:
Remove local storage discovery binary

wyc commented on pkg/printers/internalversion/describe.go:
```diff
@@ -628,6 +628,8 @@ func describeVolumes(volumes []api.Volume, w *PrefixWriter, space string) {
 			printPortworxVolumeSource(volume.VolumeSource.PortworxVolume, w)
 		case volume.VolumeSource.ScaleIO != nil:
 			printScaleIOVolumeSource(volume.VolumeSource.ScaleIO, w)
+		case volume.VolumeSource.ScaleIO != nil:
```
This looks like a typo.

msau42 committed:
Add local PV deletion in the external static provisioner

vishh committed:
fix scheduler issues

Signed-off-by: Vishnu kannan <vishnuk@google.com>

msau42 committed:
Add fsGroup in plugin and update example statefulset

vishh committed:
adding scheduler predicate checks for local storage capacity isolation

Signed-off-by: Vishnu kannan <vishnuk@google.com>

jingxu97 committed:
Enable Allocatable eviction for storage resources

Enable allocatable eviction for storage resources

jingxu97 committed:
Enable Allocatable eviction for storage resources

Enable allocatable eviction for storage resources

vishh committed:
add support for local ssds in GCE

Signed-off-by: Vishnu kannan <vishnuk@google.com>

msau42 committed:
Generated updates

msau42 committed:
Update generated client-go

msau42 committed:
Fixes and hacks to get scheduler prototype working:

1. Scheduler user needs write permissions for PVCs and PVs in all
namespaces
2. Hack to manually fill in v1 type meta so that the GetReference() call works.
Still not sure about the proper fix.
3. Don't panic on error

msau42 committed:
Update local storage provisioner README

jingxu97 committed:
Add support for local storage isolation

This PR adds support for local storage isolation

jingxu97 commented:
this is a prototype. The real PR #45686 for this feature is already merged. Close this one.
